web/src/components/Upload/Uploads/Uploads.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+
+import type { FindUploads } from 'types/graphql'
+
+import TableBuilder from 'src/components/TableBuilder/TableBuilder'
+
+import { columnDefs } from './columns'
+
+const UploadsList = ({ uploads }: FindUploads) => {
+  function toDateOrNull(dateString) {
+    return dateString ? new Date(dateString) : null
+  }
+
+  function getLaterDate(date1, date2) {
+    if (!date1) return date2
+    if (!date2) return date1
+    return date1 > date2 ? date1 : date2
+  }
+
+  /*
+    Returns an object that either contains the latest validation date or the latest invalidation date
+    (depending which validation date is later) or an empty object if no dates are found
+  */
+  function findLatestValidationEntry(upload) {
```

Comment:
Updated the logic around upload validations. This function checks both for `invalidatedAt` and `validatedAt` dates and determines which one was the latest (or returns an empty object if no dates are set).

It might be easier to condense this data and replace `invalidatedAt` and `validatedAt` (as well as `invalidatedBy` and `validatedBy`) with something like `reviewedAt` (and `reviewedBy`) and then have a boolean `isValid`. This would simplify the frontend logic considerably and reduce the complexity in the `UploadValidation` model.

Reply:
100% great idea. Lmk if you would prefer to merge the PR as-is and tackle this model change separately.